unit_test/sparse/Test_Sparse_par_ilut.hpp

```diff
@@ -0,0 +1,347 @@
+
+#include <gtest/gtest.h>
+
+using namespace KokkosSparse;
+using namespace KokkosSparse::Experimental;
+using namespace KokkosKernels;
+using namespace KokkosKernels::Experimental;
+
+typedef Kokkos::complex<double> kokkos_complex_double;
+typedef Kokkos::complex<float> kokkos_complex_float;
+
+namespace Test {
+
+template <typename scalar_t, typename lno_t, typename size_type, typename device>
+std::vector<std::vector<scalar_t>> decompress_matrix(
+  Kokkos::View<size_type*, device>& row_map,
+  Kokkos::View<lno_t*, device>& entries,
+  Kokkos::View<scalar_t*, device>& values
+                                                       )
+{
+  const auto nrows = row_map.size() - 1;
+  std::vector<std::vector<scalar_t> > result;
+  result.resize(nrows);
+  for (auto& row : result) {
+    row.resize(nrows, 0.0);
+  }
+
+  auto hrow_map = Kokkos::create_mirror_view(row_map);
+  auto hentries = Kokkos::create_mirror_view(entries);
+  auto hvalues  = Kokkos::create_mirror_view(values);
+  Kokkos::deep_copy(hrow_map, row_map);
+  Kokkos::deep_copy(hentries, entries);
+  Kokkos::deep_copy(hvalues, values);
+
+  for (size_type row_idx = 0; row_idx < nrows; ++row_idx) {
+    const size_type row_nnz_begin = hrow_map(row_idx);
+    const size_type row_nnz_end   = hrow_map(row_idx+1);
+    for (size_type row_nnz = row_nnz_begin; row_nnz < row_nnz_end; ++row_nnz) {
+      const lno_t col_idx = hentries(row_nnz);
+      const scalar_t value = hvalues(row_nnz);
+      result[row_idx][col_idx] = value;
+    }
+  }
+
+  return result;
+}
+
+template <typename scalar_t, typename lno_t, typename size_type, typename device>
+void check_matrix(
+  const std::string& name,
+  Kokkos::View<size_type*, device>& row_map,
+  Kokkos::View<lno_t*, device>& entries,
+  Kokkos::View<scalar_t*, device>& values,
+  const std::vector<std::vector<scalar_t>>& expected)
+{
+  const auto decompressed_mtx = decompress_matrix(row_map, entries, values);
+
+  const auto nrows = row_map.size() - 1;
+  for (size_type row_idx = 0; row_idx < nrows; ++row_idx) {
+    for (size_type col_idx = 0; col_idx < nrows; ++col_idx) {
+      EXPECT_NEAR(expected[row_idx][col_idx], decompressed_mtx[row_idx][col_idx], 0.01)
+        << "Failed check is: " << name << "[" << row_idx << "][" << col_idx << "]";
+    }
+  }
+}
+
+
+template <typename scalar_t>
+void print_matrix(const std::vector<std::vector<scalar_t> >& matrix)
+{
+  for (const auto& row : matrix) {
+    for (const auto& item : row) {
+      std::printf("%.2f ", item);
+    }
+    std::cout << std::endl;
+  }
+}
+
+template <typename scalar_t, typename lno_t, typename size_type,
+          typename device>
+void run_test_par_ilut() {
+  typedef Kokkos::View<size_type*, device> RowMapType;
+  typedef Kokkos::View<lno_t*, device> EntriesType;
+  typedef Kokkos::View<scalar_t*, device> ValuesType;
+  typedef Kokkos::Details::ArithTraits<scalar_t> AT;
+
+  // Simple test fixture A
+  std::vector<std::vector<scalar_t> > A = {
```

Comment:
Fixed-based test. Only passes 100% of the time if OMP_NUM_THREADS=1. Can early-exit the algorithm to check various pieces of it using the commented-out code below.